### DISCUSSION TITLE
100k write cycles

### DISCUSSION THREAD

**Discussion by nash1975** [Arnold-n/P1P2MQTT#89] [General]
I finally got writing parameters via MQTT running.
I was not aware that flushing the input buffer was crucial ( P1P2/W and empty line as payload)

Regarding the 100k write cycles for the EEPROM I have one question.
Is turning heating on/off (Z0/Z1) or increasing/decreasing the Leaving water temperature associated with a parameter write?
Or is this something that is kept in volatile memory and only stored when the heatpump is shut down?

Greetings,
Jens

**Comment by gmcmicken** [comment]
On my system there is a field setting to restore previous state after power interruption. Which I'm assuming means to write to memory or not.

**Comment by Arnold-n** [comment]
@nash1975:
> I finally got writing parameters via MQTT running. I was not aware that flushing the input buffer was crucial ( P1P2/W and empty line as payload)

This surprises me, an empty line should not be necessary (at least not for more recent versions), the ESP flushes the ATmega serial input after an ATmega reset or reflash. You have version v0.9.42 running on both, right? 

> Regarding the 100k write cycles for the EEPROM I have one question. Is turning heating on/off (Z0/Z1) or increasing/decreasing the Leaving water temperature associated with a parameter write? Or is this something that is kept in volatile memory and only stored when the heatpump is shut down?

Unfortunately I do not know the answer. Some observations:
-the Daikin manual for the Modbus interface for controlling F-series models warns that each setting change results in a flash write, and that the number of setting changes should be limited to 7000/year
-the Daikin manual for the LAN adapter or Wifi adapter for the E-series does not contain such a warning
-in my system LWT and on/off setting changes survive a power cycle, and I have not found @gmcmicken's field setting in my system

So I can only assume that
-(some) F-series do not have flash wear leveling, imposing a setting change limit
-(hopefully, more recent) E-series do have flash wear leveling, which should thus impose a much higher (less restrictive) setting change limit

@gmcmicken:
> On my system there is a field setting to restore previous state after power interruption. Which I'm assuming means to write to memory or not.

Which field setting would this be? I could add it to the README to avoid flash wear.

**Reply by gmcmicken** [reply]
For VRV and Sky Air (F series) to control auto restore state after power interruption:

12(22) - 05 - 01: No auto restore
12(22) - 05 - 02: Auto restore

I have no idea if this controls writing of the state, or if it simply controls the restoration of state. My service manual used the terminology "equipped" which makes me think it is indeed whether there is non volatile memory available. But who knows!

**Reply by nash1975** [reply]
@Arnold-n : Regarding flushing: At least afterwards it was working. I tried several days before without success. As I tried several things the success might also be related to something else I tried.

Regarding 100k write cycles.
I opened my Daikin Altherma 3 (2023 model) device and found a Renesas µC
The datasheet talks about 10k write cycles for the data flash.

There is in the area an 8pin IC.
I have not found out if this is an external EEPROM. There is no writing on the IC.


![2023-11-23 08_10_23-Clipboard](https://github.com/Arnold-n/P1P2Serial/assets/132844511/97a4adf6-0408-40df-9bfb-8d51bc965905)
![2023-11-23 08_11_53-Clipboard](https://github.com/Arnold-n/P1P2Serial/assets/132844511/4ab68fa4-6461-46ce-8aad-385a2c46511c)
![20231122_204834](https://github.com/Arnold-n/P1P2Serial/assets/132844511/2936a48b-5c0e-494d-9212-87feac1a1100)

**Comment by nicx** [comment]
very interesting discussion... I ran into the write cycle buffer problem, too. (see https://github.com/Arnold-n/P1P2Serial/issues/94) with my 2023 series Altherma 3 Ech2o.

another question I am asking myself: are parameters set by the Madoka handled in the same way? So when changing the room temperature via the Madoka, will this end up in a write cycle, too?

**Reply by Arnold-n** [reply]
I think every Madoka setting results in a write action. Although we have no confirmation, it would surprise me if the Daikin models do not have any wear leveling, so the problem should be rather theoretical for E-series Daikins. The only real warning I found applied to F-series models with much simpler protocols.
